restconnector/detail/RestZeqTranslator.cpp

```diff
@@ -23,54 +25,67 @@ RestZeqTranslator::~RestZeqTranslator()
 {
 }
 
-::zeq::Event RestZeqTranslator::translate( const std::string& body ) const
+zeq::Event RestZeqTranslator::translate ( const std::string& request, const std::string& body ) const
 {
-    //Here we should use the rest2Zeq mapper that need to be created in zeq::vocabulary
-    if( command_ == "setCamera" )
-        // This must be in the vocabulary
-        // The vacabulary is responsible for returning the
-        // zeq event
-        return zeq::vocabulary::serializeJSON( zeq::hbp::EVENT_CAMERA, body );
-    if( command_ == "request" )
-        //  This must be in the vocabulary
-        return zeq::vocabulary::serializeRequest( zeq::hbp::EVENT_IMAGEJPEG );
-    return zeq::vocabulary::serializeEcho("Unknown REST command");
+    const std::string& command = getCommand_( request );
+
+    std::map< std::string, lunchbox::uint128_t >::const_iterator it = vocabularySubscribed_.find( command );
+    if( it == vocabularySubscribed_.end() )
+        LBTHROW(CommandNotFound( command ) );
+
+    return zeq::vocabulary::serializeJSON( it->second, body );
 }
 
-bool RestZeqTranslator::isCommandSupported( const std::string& request )
+zeq::Event RestZeqTranslator::translate( const std::string& request ) const
 {
-    lunchbox::Strings dataParts;
+    const std::string& command = getCommand_( request );
+
+    std::map< std::string, lunchbox::uint128_t >::const_iterator it = vocabularyPublished_.find( command );
+    if( it == vocabularyPublished_.end() )
+        LBTHROW(CommandNotFound( "Error: " + command ) );
+
+    return zeq::vocabulary::serializeRequest( it->second );
+}
 
+std::string RestZeqTranslator::getCommand_( const std::string& request ) const
+{
     if( request.empty() )
-    {
-        LBINFO << "Error: The request string is empty." << std::endl;
-        return false;
-    }
+        LBTHROW(InvalidRequest("The request string is empty") );
+
+    lunchbox::Strings dataParts;
 
     boost::split( dataParts, request, boost::is_any_of( "/" ) );
-    std::vector<std::string> commandAndPayload;
+    lunchbox::Strings commandAndPayload;
     boost::split( commandAndPayload, dataParts.back(), boost::is_any_of( "?" ) );
 
-    if( commandAndPayload.empty() )
-        return false;
+    if( ( commandAndPayload.empty() ) || ( commandAndPayload[0] == "") )
```

Comment:
commandAndPayload[0] is a char. How can this be compared to char* string literal, and why?